webapi/src/main/scala/org/knora/webapi/responders/v2/ValuesResponderV2.scala

```diff
@@ -193,13 +193,23 @@ class ValuesResponderV2(responderData: ResponderData) extends Responder(responde
              )
 
         // If this is a list value, check that it points to a real list node.
-
         _ <- submittedInternalValueContent match {
                case listValue: HierarchicalListValueContentV2 =>
                  ResourceUtilV2.checkListNodeExists(listValue.valueHasListNode, appActor)
                case _ => FastFuture.successful(())
              }
 
+        //  If this is a list value, check if list value is not root list node.
+        _ <- submittedInternalValueContent match {
+               case listValue: HierarchicalListValueContentV2 =>
+                 ResourceUtilV2.checkIfNodeIsRoot(
```

Comment:
How this name is misleading? It says what it does = checks if the list node is root. To make it more precise it might be adjusted to `checkIfNodeIsRootAndThrowsErrorIfNot`, but IMO this is not really necessary here.

On the other comment you're right, this two queries are not optimal solution. I was actually thinking to use the one I created  -  which uses inside `getListNode`. At the end it gets the node, so if it isn't in the database should also return that information.